docs/pages/preview/responsive.tsx

```diff
@@ -0,0 +1,400 @@
+import { ControlGroup } from '@ag.ds-next/react/control-group';
+import {
+	boxPalette,
+	mapSpacing,
+	tokens,
+	useWindowSize,
+} from '@ag.ds-next/react/core';
+import { Flex } from '@ag.ds-next/react/flex';
+import { Heading } from '@ag.ds-next/react/heading';
+import { AlertFilledIcon, ArrowLeftIcon } from '@ag.ds-next/react/icon';
+import { Radio as AgDSRadio } from '@ag.ds-next/react/radio';
+import { Select as AgDSSelect } from '@ag.ds-next/react/select';
+import { Stack } from '@ag.ds-next/react/stack';
+import { Text } from '@ag.ds-next/react/text';
+import { DocumentTitle } from '../../components/DocumentTitle';
+import { responsivePreviewQueryKeys } from '../../components/code/ResponsivePreviewLink';
+import { useGetScrollbarSizes } from '../../lib/hooks/useGetScrollbarSizes';
+import { useIsIos } from '../../../packages/react/src/combobox/utils';
+
+const isLocal =
+	!process.env.NEXT_PUBLIC_BASE_PATH && process.env.NODE_ENV !== 'production';
+const trustedUrls = [
+	process.env.NEXT_PUBLIC_SITE_URL,
+	process.env.NEXT_PUBLIC_PLAYROOM_URL?.startsWith('/')
+		? undefined
+		: process.env.NEXT_PUBLIC_PLAYROOM_URL,
+].filter(Boolean);
+
+const screenSizes = {
+	xs: {
+		label: '320px (xs)',
+		width: 320,
+	},
+	sm: {
+		label: `${tokens.breakpoint.sm}px (sm)`,
+		width: tokens.breakpoint.sm,
+	},
+	md: {
+		label: `${tokens.breakpoint.md}px (md)`,
+		width: tokens.breakpoint.md,
+	},
+	lg: {
+		label: `${tokens.breakpoint.lg}px (lg)`,
+		width: tokens.breakpoint.lg,
+	},
+	xl: {
+		label: `${tokens.breakpoint.xl}px (xl)`,
+		width: tokens.breakpoint.xl,
+	},
+	xxl: {
+		label: `${tokens.breakpoint.xxl}px (xxl)`,
+		width: tokens.breakpoint.xxl,
+	},
+};
+type Sizes = keyof typeof screenSizes;
+
+const constructRadioId = (size: Sizes) => `radio-id-${size}`;
+
+type UnmountTargets = 'radio' | 'select';
+
+export default function ResponsivePage() {
+	const router = useRouter();
+	const pathname = usePathname();
+	const searchParams = useSearchParams();
+	const [frameSize, setFrameSize] = useState<Sizes>('xs');
+	const [iframeSrc, setIframeSrc] = useState<undefined | null | string>(
+		undefined
+	);
+	const iframeRef = useRef<HTMLIFrameElement>(null);
+	const selectRef = useRef<HTMLSelectElement>(null);
+	const scrollbarSizes = useGetScrollbarSizes();
+	const isIos = useIsIos();
+
+	useEffect(() => {
+		const initQueryParams = new URLSearchParams(window.location.search);
+		const defaultSize = initQueryParams.get(
+			responsivePreviewQueryKeys.previewSize
+		);
+
+		if (defaultSize && defaultSize in screenSizes) {
+			setFrameSize(defaultSize as Sizes);
+		}
+	}, [setFrameSize]);
+
+	useEffect(() => {
+		const initQueryParams = new URLSearchParams(window.location.search);
+		const iframeSrc = initQueryParams.get(responsivePreviewQueryKeys.frameSrc);
+
+		// Verify address is from AgDS
+		if (iframeSrc) {
+			try {
+				const encodedUrl = encodeURI(iframeSrc);
+
+				// If URL is AgDS, allow through
+				if (encodedUrl.startsWith('/')) {
```

Comment:
I don't think it's an issue, but I was bit taken aback by `/foo` loading our doc's 404 page in the iframe. The only other option is to add all of the allowed paths (/patterns, /yourgov etc), but that feels like overkill